### DISCUSSION TITLE
Update the dependencies used by the site layout

### DISCUSSION THREAD

**Discussion by duboisp** [gc-da11yn/gc-da11yn.github.io#67] [Development (Open discussions)]
I would like to suggest to replace the local dependencies of wet-boew and gcweb by the files hosted on the Canada.ca CDN.

Change the line 45-46 in the [basic layout](https://github.com/gc-da11yn/gc-da11yn.github.io/blob/main/src/_includes/layouts/base.njk#L45-L46) which is:
```xml
	<script src="{{ rootPath }}wet-boew/js/wet-boew.min.js"></script>
	<script src="{{ rootPath }}gcweb/js/theme.min.js"></script>
```
by
```xml
	<script src="https://www.canada.ca/etc/designs/canada/wet-boew/js/wet-boew.min.js"></script>
	<script src="https://www.canada.ca/etc/designs/canada/wet-boew/js/theme.min.js"></script>
```

Idem for the CSS file included in the [partial file head.njk at line 14](https://github.com/gc-da11yn/gc-da11yn.github.io/blob/main/src/_includes/partials/head.njk#L14) to be replaced by `https://www.canada.ca/etc/designs/canada/wet-boew/css/theme.min.css`

From this project we are not going to create our own version of wet-boew or of GCWeb So instead of maintaining a dependencies, Soon wet-boew and gcweb are going to be released 2-3 times per months. I recommend to point to the latest version hosted in Canada.ca. We are not planning to do any breaking changes in wet-boew and gcweb from the web authors point of view.

That don't block us to extend the existing style, like it was already configured neither to submit a PR in wet-boew or GCWeb GitHub repository to fix an accessibility issue.

I will be interested to know why we need to have a custom CSS for this site. Is there missing a feature / a style in GCWeb? Also that information would be useful later for the GC Design System project.

**Comment by shawnthompson** [comment]
1. Like all others who use WET-BOEW, I fear that an update to that repo might break something we are using. It probably won't but that could still happen, joys of dependencies.  

> We are not planning to do any breaking changes in wet-boew and gcweb from the web authors point of view.

That doesn't mean it won't happen. And if one of those changes is causes an accessibility issue, not planned, I would hate to have a site on accessibility that doesn't have accessibility first in mind. 

2. Presently we are using custom styles for the banner, the contribute section, and the font-awesome icons spacing instead of using `mrgn-rght-md` in every class, I wanted it to be the same everywhere we use it.

Although I'd love to add the Contribute section in WET-BOEW.

**Reply by duboisp** [reply]
> Like all others who use WET-BOEW, I fear that an update to that repo might break something we are using. It probably won't but that could still happen, joys of dependencies.

You that wet-boew and gcweb are now following semantic version 2.0 and our public API is defined here: https://wet-boew.github.io/wet-boew-documentation/decision/3.html
We did that to address that specific concern. That is also why the version of GCWeb don't match anymore the version of wet-boew.

> That doesn't mean it won't happen

I don't understand your trust issue here. I implemented a review mechanism in wet-boew and gcweb to avoid that kind of error. Also, we worked hard on improving the release note to be clear and descriptive of what the changes is. 

I can understand that it could be beneficial to lock at the major version when the site are self managing the site template, like here.

> Presently we are using custom styles for the banner, the contribute section, and the font-awesome icons spacing instead of using mrgn-rght-md in every class, I wanted it to be the same everywhere we use it.

Ok, I will review it and move those CSS in GCWeb instead.

**Reply by shawnthompson** [reply]
Do we want to move the CSS to GCWeb or WET? I would like to see the base use it. Because I'm still waiting for the Design team to officially approve it. And I'd also like to create different variants.

**Reply by duboisp** [reply]
It will most likely to be moved in GCWeb. I am currently working on making GCWeb more independent from wet-boew to optimize it like avoiding multiple override.

> Because I'm still waiting for the Design team to officially approve it.

What design team you are waiting for? 

> And I'd also like to create different variants.

Variant of the theme or variant as CMS?

**Reply by shawnthompson** [reply]
I tagged @pcwsmith in this discussion https://github.com/gc-da11yn/gc-da11yn.github.io/discussions/64.

I also spoke to @delisma in slack about it.

Maybe you can bring it up during your meeting with them.

I'd really like to create a standards for projects that are open source and a variant for the different platforms,  like GIthub, Gitlab, Gitbucket and any other ones I might be forgetting. 

I really like it the way it is and to make it stick out that it's an open source project. I'm scared it gets buried in the footer menu or even worse in the main menu.

**Reply by duboisp** [reply]
Yes I agree that will be nice if guidance was provided on how to make website for open source project. We are still waiting for guidance on how to make website for guidance (like a simple top banner).

**Reply by shawnthompson** [reply]
Related to: #64

**Comment by shawnthompson** [comment]
Let's vote on it at the next dev meeting.

**Comment by shawnthompson** [comment]
Will be moving to CDN and taking the WET dependency out of the `package.json`.
